crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -5,6 +5,16 @@ use std::collections::HashMap;
 use std::sync::{Arc, RwLock, Weak};
 use std::time::{Duration, Instant};
 
+#[cfg_attr(test, mockall::automock)]
+pub trait Metrics: Send + Sync + 'static {
+    fn native_price_cache(&self, misses: usize, hits: usize);
+}
+
+struct NoopMetrics;
```

Comment:
For some reason, I think we may already have some `NoopMetrics` type somewhere... Might be useful to reuse that one (if I'm not remembering incorrectly).

Reply:
There is currently only a `NoopMetrics` struct in the `solver` crate.
We could move that into `shared/src/metrics` and use that across all crates.

Reply:
Ah OK. Here is fine then.